### ISSUE TITLE
AlphaZero: deploy best checkpoint as 'the-oracle' ONNX personality

### ISSUE THREAD

**Issue by edetruth** [edetruth/shanghai-tracker#24]
## Goal

Export the best trained AlphaZero checkpoint to ONNX and wire it up as the `the-oracle` AI personality in the web app, alongside the existing `the-mastermind`.

## Current state

- `ml/pimc/alphazero/export.py` exists and works (opset 12, tested)
- Web app has full game UI and `AIPersonality` type in `src/game/types.ts`
- `src/game/ai.ts` has `AI_EVAL_CONFIGS` keyed by personality — `the-oracle` not yet added
- v3 best checkpoint is at `ml/pimc/alphazero/checkpoints_v3/best.pt` (275 avg, 34.5% win rate — functional but below PIMC)
- v4 training running — wait for result before exporting

## Steps

1. **Export** — once v4 (or a better run) finishes:
   ```bash
   cd D:/shanghai-tracker/ml/pimc
   python -m alphazero.export \
       --checkpoint alphazero/checkpoints_vN/best.pt \
       --output alphazero/exports/shanghai_oracle.onnx
   ```

2. **Bundle** — copy ONNX to `public/models/shanghai_oracle.onnx` or serve via API

3. **Runtime** — add ONNX Runtime Web to the app, load model, wire inference to the 5 action heads (discard/draw/buy/laydown/value)

4. **Personality** — add `the-oracle` to `AIPersonality` union type and `AI_EVAL_CONFIGS` in `src/game/ai.ts`

## Blocker

Waiting on v4 training result. If v4 doesn't improve significantly over v3 (275 avg), consider resolving #23 first (mixed PIMC opponents) before exporting.

## Notes

- State vector is 170-dim (documented in `ml/pimc/alphazero/network.py`)
- ONNX input: `float32[1, 170]`, outputs: discard_logits, draw_logits, buy_logit, laydown_logit, value
- opset 12 chosen for broad browser ONNX Runtime Web compatibility
